drivers/mtd_spi_nor/mtd_spi_nor.c

```diff
@@ -539,6 +543,9 @@ static int mtd_spi_nor_power(mtd_dev_t *mtd, enum mtd_power_state power)
     switch (power) {
         case MTD_POWER_UP:
             mtd_spi_cmd(dev, dev->params->opcode->wake);
+#if defined(MODULE_XTIMER)
+            xtimer_usleep(dev->params->wait_chip_wake_up);
```

Comment:
I'd rather avoid adding more config options. Just poll the JDEC ID register every 1ms until it returns something other than 0. (or was it `0xFF`?)

Those wake-up times can vary quite a bit between different flash chips, and I'd rather avoid having to tweak the board configuration each time the flash chip changes.

Reply:
> I'd rather avoid adding more config options. Just poll the JDEC ID register every 1ms until it returns something other than 0. (or was it 0xFF?)

Sure but you would need a timeout in that case, do you have a suggested value?

Reply:
In a different project I would poll the flash 255 times with a wait time of 1ms in between. If it didn't wake up by then it was probably broken.

Reply:
I think it is okay to add another parameter to the param struct. All of these parameters can be copied directly from the datasheet of the flash chip. 

I would be fine with a simple timeout if this is only used in the init of the chip, but the current code has potential to be used in the hot path of upper FS layers. I think it is better to have a well-defined sleep here instead of adding additional SPI reads and wakeups here.

Reply:
But Flash chips are easily interchangeable.
You might swap them out with a new board revision without having to change the software - all the information you need can be retrieved from the flash chip at run-time.

Reply:
> But Flash chips are easily interchangeable.

But then the parameters would have to be changed anyway…

>  You might swap them out with a new board revision without having to change the software - all the information you need can be retrieved from the flash chip at run-time.

I'm fine with (optionally) retrieving all timings from the chip at boot after a wake with your suggestion, but I rather not have a loop in the normal wake function.

Reply:
So what is the conclusion here? I re-charged my pinetime so could address the comments next weekend.

Reply:
> But then the parameters would have to be changed anyway…

Not really, `wait_chip_erase` & co are only lower bounds.
The driver will still work if the chip has different timings.

Reply:
Do we know common wakeup time values?

Reply:
Ok, so we had a lengthy discussion in matrix ending up with the realization that without measurements, we cannot conclude.

We discussed these options:
1. poll every 1ms (up to N times) (1ms configurable)
2. poll every params->wait ms (up to N times)
3. wait params->wait ms once, check

1. was the simplicity and error case winner. 3. is the power and "it's already there" winner. The power argument cannot be quantified. @fjmolinas expressed leaning towards 3. We agreed that this PR should be unblocked.

@benpicco would you be fine with going with that (3., adding a check if the device actually woke up)?

Reply:
Why not 2 then?
That's how all the other timeouts are handled anyway. This way the code doesn't break if a flash chip with different characteristics is used.